pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentSizeBasedFlushThresholdUpdater.java

```diff
@@ -102,6 +102,7 @@ public synchronized void updateFlushThreshold(PartitionLevelStreamConfig streamC
     // less same characteristics at any one point in time).
     // However, when we start a new table or change controller mastership, we can have any partition completing first.
     // It is best to learn the ratio as quickly as we can, so we allow any partition to supply the value.
+    // FIXME: The stream may not have partition "0"
```

Comment:
Will it work fine without this fix?

Reply:
Segment size based threshold won't work, but rows/time will work. We can fix that separately.

Reply:
Sure.